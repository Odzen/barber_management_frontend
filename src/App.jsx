import { Routes, Route } from 'react-router-dom'
import PropTypes from 'prop-types'
import MainRouter from './containers/mainRouter/index'
import 'antd/dist/reset.css'

const App = ({ location }) => {
  return (
    /*MainRouter contains our main routes*/
    <Routes>
      <Route path='/*' element={<MainRouter location={location} />} />
    </Routes>
  )
}

App.propTypes = {
  location: PropTypes.object.isRequired
}

export default App
