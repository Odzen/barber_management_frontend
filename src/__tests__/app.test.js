// import { render, screen, cleanup, fireEvent } from '@testing-library/react'
import { MemoryRouter, Routes, Route } from 'react-router-dom'
import App from '../App'
import renderer from 'react-test-renderer'

describe('testing to App', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn()
      }))
    })
  })

  test('should return App Component', () => {
    let app = renderer
      .create(
        <MemoryRouter initialEntries={['/users/mjackson']}>
          <App location={{ pathname: '/' }} />
        </MemoryRouter>
      )
      .toJSON()
    expect(app).toMatchSnapshot()
  })
})
