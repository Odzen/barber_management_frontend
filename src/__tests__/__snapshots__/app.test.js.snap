// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing to App should return App Component 1`] = `
<section
  className="background h-100"
>
  <div
    className="h-100"
  >
    <div
      className="row g-0 align-items-center justify-content-center h-100 px-5"
    >
      <div
        className="col-xxl-3 col-xl-5 col-lg-5 col-md-7 col-sm-9 col-lg-auto sw-lg-70"
      >
        <div
          className="card shadow-lg"
        >
          <div
            className="card-body p-3"
          >
            <div
              className="d-flex justify-content-center align-items-center"
            >
              <div
                className="d-flex container-title p-3 m-2"
              >
                <img
                  alt=""
                  className="imgIcon"
                  src={
                    {
                      "process": [Function],
                    }
                  }
                />
                <span
                  className="fs-4 card-title fw-bold"
                >
                  Bienvenido
                </span>
                <img
                  alt=""
                  className="imgIcon"
                  src={
                    {
                      "process": [Function],
                    }
                  }
                />
              </div>
            </div>
            <div
              className="d-flex justify-content-center align-items-center m-3"
            >
              <form
                className="ant-form ant-form-horizontal css-dev-only-do-not-override-zjzpde login-form"
                id="normal_login"
                onReset={[Function]}
                onSubmit={[Function]}
              >
                <div
                  className="ant-form-item css-dev-only-do-not-override-zjzpde"
                >
                  <div
                    className="ant-row ant-form-item-row css-dev-only-do-not-override-zjzpde"
                    style={{}}
                  >
                    <div
                      className="ant-col ant-form-item-control css-dev-only-do-not-override-zjzpde"
                      style={{}}
                    >
                      <div
                        className="ant-form-item-control-input"
                      >
                        <div
                          className="ant-form-item-control-input-content"
                        >
                          <span
                            className="ant-input-affix-wrapper css-dev-only-do-not-override-zjzpde"
                            onClick={[Function]}
                          >
                            <span
                              className="ant-input-prefix"
                            >
                              <span
                                aria-label="user"
                                className="anticon anticon-user site-form-item-icon"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="user"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                                  />
                                </svg>
                              </span>
                            </span>
                            <input
                              aria-required="true"
                              className="ant-input css-dev-only-do-not-override-zjzpde"
                              disabled={false}
                              hidden={null}
                              id="normal_login_email"
                              name="email"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              placeholder="Correo"
                              type="text"
                              value=""
                            />
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="ant-form-item css-dev-only-do-not-override-zjzpde"
                >
                  <div
                    className="ant-row ant-form-item-row css-dev-only-do-not-override-zjzpde"
                    style={{}}
                  >
                    <div
                      className="ant-col ant-form-item-control css-dev-only-do-not-override-zjzpde"
                      style={{}}
                    >
                      <div
                        className="ant-form-item-control-input"
                      >
                        <div
                          className="ant-form-item-control-input-content"
                        >
                          <span
                            className="ant-input-affix-wrapper css-dev-only-do-not-override-zjzpde"
                            onClick={[Function]}
                          >
                            <span
                              className="ant-input-prefix"
                            >
                              <span
                                aria-label="lock"
                                className="anticon anticon-lock site-form-item-icon"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="lock"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                                  />
                                </svg>
                              </span>
                            </span>
                            <input
                              aria-required="true"
                              className="ant-input css-dev-only-do-not-override-zjzpde"
                              disabled={false}
                              hidden={null}
                              id="normal_login_password"
                              name="password"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              placeholder="ContraseÃ±a"
                              type="password"
                              value=""
                            />
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="ant-form-item css-dev-only-do-not-override-zjzpde"
                  style={
                    {
                      "color": "white",
                      "marginBottom": "0",
                    }
                  }
                >
                  <div
                    className="ant-row ant-form-item-row css-dev-only-do-not-override-zjzpde"
                    style={{}}
                  >
                    <div
                      className="ant-col ant-form-item-control css-dev-only-do-not-override-zjzpde"
                      style={{}}
                    >
                      <div
                        className="ant-form-item-control-input"
                      >
                        <div
                          className="ant-form-item-control-input-content"
                        >
                          <button
                            className="ant-btn css-dev-only-do-not-override-zjzpde ant-btn-primary login-form-button btn-primary"
                            disabled={false}
                            onClick={[Function]}
                            type="submit"
                          >
                            <span>
                              Ingresar
                            </span>
                          </button>
                          O
                           
                          <a
                            className=""
                            onClick={[Function]}
                          >
                            Registrarse
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="loading"
                >
                  <div
                    aria-busy={true}
                    aria-live="polite"
                    className="ant-spin ant-spin-spinning css-dev-only-do-not-override-zjzpde"
                  >
                    <span
                      aria-label="loading"
                      className="anticon anticon-loading anticon-spin ant-spin-dot"
                      role="img"
                      style={
                        {
                          "fontSize": 24,
                        }
                      }
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="loading"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="0 0 1024 1024"
                        width="1em"
                      >
                        <path
                          d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </form>
            </div>
          </div>
          <div
            className="card-footer py-3 border-0"
          >
            <div
              className="text-center text-white"
            >
              Todos los derechos reservados Â© 2023
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
